app/main.py

```diff
@@ -1,3 +1,62 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import os
+
+from app.car import Car
+from app.shop import Shop
+from app.customer import Customer
+
+
+def shop_trip() -> None:
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    config_path = os.path.join(current_dir, "config.json")
+
+    with open(config_path, "r") as config_file:
+        config = json.load(config_file)
+
+    fuel_price = config["FUEL_PRICE"]
+    customers_data = config["customers"]
+    shops_data = config["shops"]
+
+    customers = []
+    for customer_data in customers_data:
+        car = Car(customer_data["car"]["fuel_consumption"])
+        customer = Customer(
+            customer_data["name"],
+            customer_data["location"],
+            customer_data["money"],
+            customer_data["product_cart"],
+            car
+        )
+        customers.append(customer)
+
+    shops = []
+    for shop_data in shops_data:
+        shop = Shop(
+            shop_data["name"],
+            shop_data["location"],
+            shop_data["products"]
+        )
+        shops.append(shop)
```

Comment:
use list comprehension here 
and unpacking  
```suggestion
    shops = []
    for shop_data in shops_data:
        shop = Shop(**shop_data)
        shops.append(shop)
```

Reply:
Done